### PR TITLE
Add regex test for parsing runtime-assets expressions

### PR DESCRIPTION
**Pull request by stephentoub**
cc: @joperezr, @danmoseley

### CODE REVIEW

File: src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs
Comment (danmoseley) [inline src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs:1458]: where is this Equals getting called?

Comment (stephentoub) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs:1458]: By Distinct()

File: src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs
Comment (danmoseley) [inline src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs:1425]: is there anything useful we can do with the regex here? I guess I can't think of much unless you want to validate the ToString() == r.Pattern and Options = r.Options.

**Review by danmoseley — APPROVED** [review]

File: src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs
Comment (danmoseley) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs:1458]: oh yes. are there intentionally dupes in there?

Comment (stephentoub) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs:1458]: A non-trivial number of patterns show up multiple times with distinct options, most commonly one with Compiled and one without.

File: src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs
Comment (stephentoub) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs:1425]: I was thinking we might at least want to run IsMatch to at least force some portion of the code to execute, and just validate it successfully JITs / etc.  Beyond that, I'm not sure.  I expect subsequently we'll be looking at generating inputs for at least a subset of the expressions and then validating those inputs here.

File: eng/Versions.props
Comment (joperezr) [inline eng/Versions.props:130]: I believe you also need to add a line in Version.Details.xml file in order to have this one update automatically via Darc.  You can see an example for the one above here:

[https://github.com/dotnet/runtime/blob/c9f9697fe7d10c0c1d3f17cb258518bc8d4e7cc6/eng/Version.Details.xml#L125-L128](https://github.com/dotnet/runtime/blob/c9f9697fe7d10c0c1d3f17cb258518bc8d4e7cc6/eng/Version.Details.xml#L125-L128)

Just copy and paste that and adjust the Name.

Comment (stephentoub) [reply eng/Versions.props:130]: Ah, cool, thanks.

File: src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs
Comment (joperezr) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs:1425]: Yeah, I was talking offline with @danmoseley and it would be good if we add to that json some sample input that should match (perhaps also include the expected catpure groups too), and some that should not match. We then later can share this json with other engines as a test bed.

**Review by joperezr — APPROVED** [review]

File: eng/Versions.props
Comment (joperezr) [reply eng/Versions.props:130]: Given CI is almost done, it's fine if you want to address this in a subsequent PR. Or I can do that too on mine once I rebase after you merge the code cleanup one.

Comment (stephentoub) [reply eng/Versions.props:130]: It's ok.  I actually want to wait until after #62318 is merged, and then do an outerloop run with that before merging this.

**Pull request merged**
